diff --git a/node_modules/react-native-worklets-core/cpp/WKTJsiWorklet.h b/node_modules/react-native-worklets-core/cpp/WKTJsiWorklet.h
index ea82623..9e7dc52 100644
--- a/node_modules/react-native-worklets-core/cpp/WKTJsiWorklet.h
+++ b/node_modules/react-native-worklets-core/cpp/WKTJsiWorklet.h
@@ -6,6 +6,7 @@
 #include <string>
 #include <utility>
 #include <vector>
+#include <cctype> 
 
 #include "WKTJsiHostObject.h"
 #include "WKTJsiWorkletContext.h"
@@ -303,7 +304,7 @@ private:
     }
     
     // Double-check if the code property is valid.
-    bool isCodeEmpty = std::all_of(_code.begin(), _code.end(), std::isspace);
+    bool isCodeEmpty = std::all_of(_code.begin(), _code.end(), [](char c) { return std::isspace(static_cast<unsigned char>(c)); });
     if (isCodeEmpty) {
       std::string error = "Failed to create Worklet, the provided code is empty. Tips:\n"
         "* Is the babel plugin correctly installed?\n"
